src/app/services/folder.js

```diff
@@ -160,37 +160,67 @@ module.exports = (Model, App) => {
     return treeSize;
   };
 
-  // A tree without using hierarchy library
   const GetTree = async (user, rootFolderId = null) => {
-    const folderContents = (
-      await Model.folder.findOne({
-        where: { id: { [Op.eq]: rootFolderId || user.root_folder_id } },
-        include: [
-          {
-            model: Model.folder,
-            as: 'children',
-            include: [
-              {
-                model: Model.file,
-                as: 'files',
-              },
-            ],
-          },
-          {
-            model: Model.file,
-            as: 'files',
-          },
-        ],
-      })
-    ).toJSON();
+    const rootElements = [];
+    const pendingFolders = [{
+      folderId: rootFolderId || user.root_folder_id,
+      elements: rootElements
+    }];
+
+    while (pendingFolders.length) {
+      const { folderId, elements } = pendingFolders.shift();
+      const folder = await getFolderByFolderId(folderId);
+      folder.files = await getFilesByFolderId(folderId);
+      folder.children = [];
+
+      const folders = await getChildrenFoldersByFolderId(folderId);
+
+      pendingFolders.push(
+        ...folders.map(f => {
```

Comment:
Is the spread strictly necessary here? 

Could the code work without the spread or using `.concat`? As is not being paginated, this could be an overhead.

Reply:
Right! Actually since im iterating the folders inside, i can add the items one be one on creation time, and so we dont need the spread! :pray: